src/sage/combinat/sf/sfa.py

```diff
@@ -417,6 +417,26 @@ def is_integral_domain(self, proof=True):
             """
             return self.base_ring().is_integral_domain()
 
+        def fraction_field(self):
+            r"""
+            Return the fraction field of ``self``.
+
+            EXAMPLES::
+
+                sage: s = SymmetricFunctions(QQ).s()
+                sage: s.fraction_field()
+                Fraction Field of Symmetric Functions over Rational Field in the Schur basis
+            """
+            if not self.is_integral_domain():
+                raise TypeError("self must be an integral domain.")
```

Comment:
```suggestion
                raise TypeError("self must be an integral domain")
```
to follow Python's conventions. (Should also be changed in the poly rings at some point.)